### PR TITLE
Integrate cJSON

### PR DESCRIPTION
**Pull request by randy408**
[cJSON](https://github.com/DaveGamble/cJSON) an MIT-licensed JSON parser, it is used in the [ESP8266 SDK](https://github.com/espressif/ESP8266_RTOS_SDK/tree/1e294d110262625d6e8df0b4c0ceb5fe9f2f6b58/components/cjson).

Upstream PR https://github.com/DaveGamble/cJSON/pull/378

### CODE REVIEW

File: projects/cjson/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+################################################################################
+
+FROM gcr.io/oss-fuzz-base/base-builder
+
+MAINTAINER randy408@protonmail.com
+
+RUN apt-get update && apt-get install -y cmake
+
+RUN git clone --depth 1 https://github.com/DaveGamble/cJSON.git cjson
+RUN git clone --depth 1 https://github.com/google/fuzzer-test-suite
```
Comment (jonathanmetzman) [inline projects/cjson/Dockerfile]: Why clone fuzzer-test-suite?

File: projects/cjson/build.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+mkdir build
+cd build
+cmake -DBUILD_SHARED_LIBS=OFF -DENABLE_CJSON_TEST=OFF ..
+make -j5
```
Comment (jonathanmetzman) [inline projects/cjson/build.sh]: `make -j$(nproc)`

File: projects/cjson/build.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+$CXX $CXXFLAGS -std=c++11 -I. \
+    $SRC/cjson/fuzzing/cjson_read_fuzzer.cc \
+    -o $OUT/cjson_read_fuzzer \
+    $LIB_FUZZING_ENGINE $SRC/cjson/build/libcjson.a
+
+find $SRC/cjson/fuzzing/inputs -name "*" | \
+     xargs zip $OUT/cjson_read_fuzzer_seed_corpus.zip
+
+cp $SRC/cjson/fuzzing/json.dict $SRC/cjson/fuzzing/cjson_read_fuzzer.dict
```
Comment (jonathanmetzman) [inline projects/cjson/build.sh]: nit: can this be copied directly to `$OUT`?

**Comment by jonathanmetzman** [comment]
We'll accept this project pending approval from upstream.
Please update this PR once the upstream PR has been accepted so travis will run again.

**Comment by randy408** [comment]
Is building with `$CC` supported? https://github.com/DaveGamble/cJSON/blob/master/fuzzing/ossfuzz.sh#L11

**Pull request merged**